fix: Align documentation & behavior for newer REST API endpoints (#3321)
NVIDIA/bare-metal-manager-core: docs/architecture/health_aggregation.md

## Health report overrides

Site administrators can inspect and update the health state of a NICo-managed host
through these REST operations:

- `GET /v2/org/{org}/nico/machine/{machineId}/health-report` lists all reports.
- `PUT /v2/org/{org}/nico/machine/{machineId}/health-report` creates or updates an override.
- `DELETE /v2/org/{org}/nico/machine/{machineId}/health-report/{source}` removes an override.

The override API offers 2 different modes of operation:

1. `Merge` - In this mode, any health probe alerts indicated in the override
  will get merged with health probe alerts reported by built-in NICo tools in order
  to derive the aggregate host health status. **This mode is meant to augment the internal health monitoring mechanism with additional sources of health data**
2. `Replace` - In this mode, the health probe alerts reported by built-in NICo
  monitoring tools will be ignored. Only alerts that are passed as part of the
  override will be taken into account. If the override list is empty, the system

>   will behave as if the Host would be fully healthy. **This mode is meant to bypass the internal health data in case the site operator desires a different behavior**

Reviewer: I know I'm not responding to your change (which was good), but... Does a 2-item list really need numbering? I would argue for simply two paragraphs since it's not sensitive to ordering, and we already overuse bullets.

```suggestion
`Merge` - In this mode, any health probe alerts indicated in the override will get merged with health probe alerts reported by built-in NICo tools in order to derive the aggregate host health status. **This mode is meant to augment the internal health monitoring mechanism with additional sources of health data**.

`Replace` - In this mode, the health probe alerts reported by built-in NICo monitoring tools will be ignored. Only alerts that are passed as part of the override will be taken into account. If the override list is empty, the system will behave as if the Host would be fully healthy. **This mode is meant to bypass the internal health data in case the site operator desires a different behavior**.
```

The API allows multiple `Merge` overrides on a host at the same time by using a different `source` identifier for each report.

> This allows to integrate health information from multiple external systems and users which are not at risk of overriding each others data. E.g. health information from an external fleet health monitoring system and from SREs can be stored independently.

Reviewer: Again! Catching a change just below yours 🙃 
```suggestion
This allows you to integrate health information from multiple external systems and users which are not at risk of overriding each other's data. For example, health information from an external fleet health monitoring system and from SREs can be stored independently.
```
